addOns/reports/src/test/java/org/zaproxy/addon/reports/ReportApiUnitTest.java

```diff
@@ -0,0 +1,461 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.reports;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import net.sf.json.JSONObject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.model.Model;
+import org.zaproxy.zap.model.Context;
+
+class ReportApiUnitTest {
+
+    private ExtensionReports extReports;
+    private ReportApi reportApi;
+    private JSONObject params;
+    private Template template;
+    private ArgumentCaptor<ReportData> reportDataCaptor;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        extReports = mock(ExtensionReports.class);
+        reportApi = new ReportApi(extReports);
+        params = new JSONObject();
+        params.put(ReportApi.PARAM_TITLE, "Default Title");
+        params.put(ReportApi.PARAM_TEMPLATE, "traditional-html-plus");
+        template =
+                new Template(
+                        new File(
+                                "src/main/zapHomeFiles/reports/traditional-html-plus/template.yaml"));
```

Comment:
It would be better to not hardcode the prefix `"src/main/zapHomeFiles/`, the files are in the classpath so can be referenced that way e.g. `ReportApi.class.getResource("/reports/traditional-html-plus/template.yaml").toURI()`. In other places it can do something similar.